File: test/fixtures/java-unirest/redirect_command_data.java
import java.lang.Runtime;
import java.util.Scanner;
import kong.unirest.HttpResponse;
import kong.unirest.Unirest;

class Main {
    public static void main(String[] args) {
            HttpResponse<String> httpResponse
               = Unirest.post("http://localhost:28139")
            .header("Content-Type", "application/x-www-form-urlencoded")
            .body("foo&@" + exec("echo myfile.jg"))
            .asString();
    }
}

File: test/fixtures/kotlin-ktor/redirect_command_data.kt
import io.ktor.client.HttpClient
import io.ktor.client.request.header
import io.ktor.client.request.post
import io.ktor.client.request.setBody
import io.ktor.client.statement.bodyAsText
import io.ktor.http.isSuccess
import java.lang.Runtime
import java.util.Scanner
import kotlinx.coroutines.runBlocking

fun exec(cmd: String): String {
  try {
    val p = Runtime.getRuntime().exec(cmd)
    p.waitFor()
    val s = Scanner(p.getInputStream()).useDelimiter("\\A")
    return s.hasNext() ? s.next() : ""
  } catch (Exception e) {
    return ""
  }
}

val client = HttpClient()

runBlocking {
    val response = client.post("http://localhost:28139") {
        setBody("foo&@" + exec("echo myfile.jg"))
        header("Content-Type", "application/x-www-form-urlencoded")
    }
    if (!response.status.isSuccess()) throw Exception("Request failed with status code ${response.status}")

    println(response.bodyAsText())
}
